tests/stream.rs

```diff
@@ -199,3 +199,43 @@ fn wait() {
     assert_eq!(list().wait().collect::<Result<Vec<_>, _>>(),
                Ok(vec![1, 2, 3]));
 }
+
+#[test]
+fn publish() {
+    let (tx, rx) = channel::<_, u32>();
+    tx.send(Ok(1))
+        .and_then(|tx| tx.send(Ok(2)))
+        .and_then(|tx| tx.send(Ok(3)))
+        .forget();
+
+    let first = rx.publish();
+    let second = first.clone();
+    let third = first.clone();
+
+    let mut first = first.wait();
+    let mut second = second.wait();
+    let mut third = third.wait();
+
+    // First register the original one.
+    assert_eq!(first.next(), Some(Ok(1)));
+
+    // The second one now comes in - it should have been dropped
+    // for the first but will recieve the second.
+    assert_eq!(second.next(), Some(Ok(2)));
```

Comment:
With the current design it would not be possible. We would need something similar to `connect()` (http://reactivex.io/documentation/operators/connect.html) from Rx to allow an arbitary number of consumers to hook into the system before the first poll is carried out upstream.